atomisticparsers/utils/parsers.py

```diff
@@ -179,52 +179,53 @@ def parse_md_workflow(self, data: Dict[str, Any]) -> None:
         self.archive.workflow2 = sec_workflow
 
     def parse_interactions(self, interactions: List[Dict], sec_model: MSection) -> None:
-        interaction_dict = {}
-        for interaction_key in Interaction.m_def.all_quantities.keys():
-            interaction_dict[interaction_key] = np.array(
-                [interaction.get(interaction_key) for interaction in interactions],
-                dtype=object,
-            )
-        interaction_dict = {key: val for key, val in interaction_dict.items()}
-        interaction_types = (
-            np.unique(interaction_dict["type"])
-            if interaction_dict.get("type") is not None
-            else []
-        )
-        for interaction_type in interaction_types:
+        if not interactions:
+            return
+
+        def write_interaction_values(values):
```

Comment:
In general this function is now a bit confusing to me,  I think I do not yet fully understand all your motivations here. Please clarfiy.

Reply:
I made this part simpler to get rid of the mapping of the data to the interaction types, now I simply sort the interactions by type and collect the values until a new interaction type is encountered then write the interaction values. I had a temporary fix for the atom_indices and atom_labels because before, the shape (arbitrary) and type (np.object) is not consistent with the definition hence the error. The more formal fix, I think is to restructure the defintion so we get we are not compelled to have a fixed array.